src/main/java/internetshop/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -29,7 +29,7 @@ public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
 
     @Override
     public boolean deleteProduct(ShoppingCart shoppingCart, Product product) {
-        return shoppingCartDao.get(shoppingCart.getId()).get().getProducts().remove(product);
+        return shoppingCart.getProducts().remove(product);
```

Comment:
should we update our shoppingCart in Storage(Database) after we've changed it?

Reply:
yes, because without update db doesn't know about my changed.